### DISCUSSION TITLE
Is there a way to power on for the duration of the negative agile rates?

### DISCUSSION THREAD

**Discussion by deltamelter** [BottlecapDave/HomeAssistant-OctopusEnergy#285] [Ideas]
Trying to figure out how to configure target rate to take advantage of the negative agile rate.
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/27061628/96d17b83-489b-4e5c-9a8a-b3f85c235d48)

**Comment by BottlecapDave** [comment]
If you set your target rate sensor to your import meters, then it will find the times that are the cheapest (including negative rates). The sensor will turn on when it's in the cheap period and will turn off when it's out of the period. What part are you having issues with?

**Reply by deltamelter** [reply]
Is there a method to specify how cheap the cheap period should be?
Like a threshold/band for each target rate sensor?  eg these devices should only power on automatically when the rate is below 0p/kwh

**Reply by BottlecapDave** [reply]
There isn't, but you could use a condition that looks at the current rate sensor and performs this restriction.
